### ISSUE TITLE
Some lobby servers may be excluded because of ICMP failure

### ISSUE THREAD

**Issue by SadPencil** [CnCNet/xna-cncnet-client#210]
This issue is for commit 4b7172e745f0df5420f33c4cd39df29c1f4d05a9

The commit tests the latency of all lobby servers by sending them ping packet and measure the latency. And servers which latency are greater than 400ms (defined in MAXIMUM_LATENCY) or with ping failure are ignored. It causes some problems here:

- In some region where Internet connection is poor, the latency between the PC and the fastest server are greater than 400ms, in which case the user can't log in to the lobby because all the servers are ignored.
- A server may have ICMP message blocked in its firewall, while the TCP packets are not blocked. In this case, one can't ping the server, but the server is still available for IRC connection.

A great number of users who live in China meets the two conditions described above, and compared to the old DTA client in which they can successfully log in to the CnCNet lobby, they can't log in anymore after upgrading the DTA client.

So, to solve the problem, I suggest:
- Change the behavior of `GetAvailableServerList()`. The function won't exclude any servers anymore. A server which latency is greater than 400ms is now at the end of the server list, instead of being excluded.

**Comment by GraionDilach** [comment]
Do you think if #208/#209 changed something for the better? https://github.com/CnCNet/xna-cncnet-client/commit/4b7172e745f0df5420f33c4cd39df29c1f4d05a9 unintentionally caused that issue as well with the introduced DNS filtering and I wonder if that helped on the situation somewhat.

**Comment by SadPencil** [comment]
> 
> 
> Do you think if #208/#209 changed something for the better? [4b7172e](https://github.com/CnCNet/xna-cncnet-client/commit/4b7172e745f0df5420f33c4cd39df29c1f4d05a9) unintentionally caused that issue as well with the introduced DNS filtering and I wonder if that helped on the situation somewhat.

#208/#209 is a different optimization which has nothing to do with this issue. The key is, don't exclude any servers for slow pings, so that users with poor Internet connection can still log in to the lobby.

**Comment by SadPencil** [comment]
I'll show an example of solving this issue. I'm not creating a PR because the following solution is not beautifully written.

In file `DXMainClient\Online\Connection.cs`:
```csharp
private void ConnectToServer()
        {
            IEnumerable<Server> availableServerSortedList = GetAvailableServerList();
            List<Server> serverList = new List<Server>(availableServerSortedList);
            serverList.AddRange(Servers);
            foreach (Server server in serverList)

// rest lines are not changed
```

I found about ten users in China who suffer from not being able to connect to the lobby after upgrading the latest DTA client ( In fact, upgrade to Mental Omega 3.3.5 from 3.3.4 ). After replacing the DTA client executable with my modification above, all of them can log in to the lobby. Thus, this issue should be solved as it does affect lots of people.

**Comment by pzhlkj6612** [comment]
Hi @SadPencil .

> The key is, don't exclude any servers for slow pings, so that users with poor Internet connection can still log in to the lobby.

You are right. I did a bad thing in #142 and am working on it. :(

**Comment by Metadorius** [comment]
Fixed with merge of #212.
